### PR TITLE
Feat(budget): on push refactor

### PR DESCRIPTION
**Pull request by dochiasebastian**

### CODE REVIEW

File: budget-manager/src/app/app.component.ts [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
-    this._spengingService.spendingTotals$.subscribe(value => this.spendingTotals = value);
-  }
+  constructor(
+    private _spendingService: SpendingService, 
+    private _budgetService: BudgetService, 
+    private _dataService: DataService,
+  ) { }
 
   ngOnInit(): void {
     this._budgetService.getBudget().subscribe();
```
Comment (CatalinB7) [inline budget-manager/src/app/app.component.ts:29]: i thought we agreed not to subscribe anymore to any stream in this component

File: budget-manager/src/app/budget/budget.component.ts [hunk 2]
```diff
@@ -11,45 +13,71 @@ import {
...
       width: '300px',
-      data: {value: this.budget.value, plannedSaving: this.budget.plannedSaving}
+      data: { value: this.budget.value, plannedSaving: this.budget.plannedSaving }
     });
 
     dialogRef.afterClosed().subscribe(result => {
       if(result){
-        this.budget.value = result.value;
-        this.budget.plannedSaving = result.plannedSaving;
+        this.changeBudgetEvent.emit({value: result.value, plannedSaving: result.plannedSaving});
```
Comment (CatalinB7) [inline budget-manager/src/app/budget/budget.component.ts:77]: space after '{' and before '}'

**Review by CatalinB7 — APPROVED** [review]

File: budget-manager/src/app/app.component.ts [hunk 3]
```diff
@@ -1,40 +1,42 @@
...
-    this._spengingService.getSpendingList().subscribe();
+    this._spendingService.getSpendingList().subscribe();
```
Comment (CatalinB7) [inline budget-manager/src/app/app.component.ts:30]: do we still have to subscribe in this component?

Comment (dochiasebastian) [reply budget-manager/src/app/app.component.ts:30]: We need to get the data from the backend when the component loads so we do need this. I don't know if there is a more appropriate place to put them though...

**Pull request merged**